test/integration/grpc_retry_test.cpp

```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <gtest/gtest.h>
+
+#include <grpcpp/server_context.h>
+#include <grpcpp/support/status.h>
+#include "framework/test_logger.hpp"
+#include "network/impl/grpc_channel_builder.hpp"
+#include "qry_responses.pb.h"
+#include "queries.pb.h"
+#include "torii/query_service.hpp"
+#include "main/server_runner.hpp"
+
+class MockQueryService : public iroha::torii::QueryService {
+ public:
+  explicit MockQueryService(int max_attempts)
+      : iroha::torii::QueryService(
+          nullptr, nullptr, nullptr, getTestLogger("MockQueryService")),
+        max_attempts_(max_attempts) {}
+
+  grpc::Status Find(grpc::ServerContext *context,
+                    const iroha::protocol::Query *request,
+                    iroha::protocol::QueryResponse *response) override {
+    if (attempts_ < max_attempts_) {
+      attempts_++;
+      return grpc::Status(grpc::StatusCode::UNAVAILABLE, "");
```

Comment:
we could also set a message here to some constant and check it on the client's side

Reply:
Done in the latest version